src/filesystem/impls/appshell/node/CSharpWatcher.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * This copies the implementation of VSCode
+ * https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/files/node/watcher/win32/csharpWatcherService.ts
```

Comment:
I didn't copy this file, I have rewritten it to JavaScript, code uses TypeScript